### PR TITLE
Add Flow support

### PR DESCRIPTION
**Pull request by bigslycat**
Flow have a very bad Symbol support, but this is temporary (or not 😄) fixed by using string literals. For example look to [using Symbol.iterator](https://github.com/facebook/flow/blob/master/lib/core.js#L197) in arrays in core Flow lib. I propose a same way. It works:

```ts
export type ObservableCompatible<T> = {
  '@@observable'(): ObservableLike<T>,
}

export type ObservableLike<T> = {
  subscribe(
    onNext: (T) => mixed,
    onError?: (Error) => mixed,
    onComplete?: () => mixed,
  ): SubscriptionLike,
  subscribe(observer: Observer<T>): SubscriptionLike,
  '@@observable'(): ObservableLike<T>,
}
```

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -13,16 +13,18 @@
   },
   "scripts": {
     "test": "npm run build && mocha && tsc && node ./ts-test/test.js && check-es3-syntax -p lib/ --kill",
-    "build": "babel es --out-dir lib",
+    "build": "babel es --out-dir lib && cp es/index.js.flow lib/",
```
Comment (TrySound) [inline package.json:16]: I don't think `es/index.js.flow` is need. When you import from `symbol-observable` package flow always resolves to main field.

Comment (bigslycat) [reply package.json:16]: This may be used as `import $$observable from 'symbol-observable/es'`.

Comment (TrySound) [reply package.json:16]: Why? Bundlers will resolve `module` field.

Comment (bigslycat) [reply package.json:16]: Bundlers. Not Node.js. Check it just now:

`index.mjs`:
```js
import 'test'
```

`node_modules/test/package.json`:
```json
{
  "main": "cjs.js",
  "module": "esm.mjs"
}
```

`node_modules/test/cjs.js`:
```js
console.log('This is commonjs')
```

`node_modules/test/esm.mjs`:
```js
console.log('This is ES module')
```

Execute
```sh
node --experimental-modules index.mjs
```

Comment (TrySound) [reply package.json:16]: index.js/index.mjs pair will solve the problem. However it's still too tricky.
But okay, whatever works for you.

**Comment by benlesh** [comment]
After coming to look at this several times, I'm not sure I want to add Flow support at this time. It's a tough decision, but I'm not sure Flow has the level of popularity that I'd want to invest time in maintaining this.

**Pull request closed without merging**